tests/admin/valuesets/valueSetCreation.spec.ts

```diff
@@ -0,0 +1,347 @@
+
+    // Add concept and verify it's properly added
+    await page.getByRole("button", { name: "+ Add Concepts" }).first().click();
+    const conceptCode = "73211009"; // Diabetes mellitus SNOMED code
+    await page
+      .getByRole("textbox", { name: "Code *" })
+      .first()
+      .fill(conceptCode);
+
+    // Set status
+    await page.getByRole("combobox", { name: "Status" }).click();
+    await page.getByRole("option", { name: "Active" }).click();
+
+    // Verify concept is properly set before saving
+    await expect(
+      page.getByRole("textbox", { name: "Code *" }).first(),
+    ).toHaveValue(conceptCode);
+
+    // Test ValueSet Preview
+    await page.getByRole("button", { name: "Value Preview" }).click();
+
+    // Verify preview dialog opened
+    const previewDialog = page
+      .locator('[data-testid="preview-dialog"]')
+      .first();
+    await expect(previewDialog).toBeVisible();
+
+    // Verify specific codes appear in preview
+    await expect(previewDialog.getByText(conceptCode)).toBeVisible();
+
+    // Close preview and save
+    await previewDialog.getByRole("button", { name: /close|back/i }).click();
+    await page.getByRole("button", { name: "Save" }).click();
+
+    // Verify success
+    await expect(page.getByText("ValueSet created successfully")).toBeVisible({
+      timeout: 10000,
+    });
+  });
+
+  test("should create valueset with include rules - any system with filters", async ({
+    page,
+  }) => {
+    const uniqueSuffix = faker.string.alphanumeric(8).toLowerCase();
+    const uniqueName = `Filter Based ValueSet ${uniqueSuffix}`;
+    const uniqueSlug = `filter-based-${uniqueSuffix}`;
+
+    // Fill basic info
+    await page.getByRole("textbox", { name: "Name *" }).fill(uniqueName);
+    await page.getByRole("textbox", { name: "Slug *" }).fill(uniqueSlug);
+
+    // Add include rule with filter
+    await page
+      .getByRole("button", { name: "+ Add Rule", exact: true })
+      .first()
+      .click();
+    await page.getByRole("combobox", { name: "System" }).first().click();
+    await page.getByRole("option", { name: "SNOMED CT" }).click();
+
+    // Add filter instead of concepts
+    await page.getByRole("button", { name: "+ Add Filter" }).first().click();
+
+    // Fill filter details - use specific known properties
+    await page.getByRole("combobox", { name: "Property" }).first().click();
+    await page.getByRole("option", { name: "concept" }).first().click();
+
+    await page.getByRole("combobox", { name: "Operator" }).first().click();
+    await page.getByRole("option", { name: "equals" }).click();
+
+    await page.getByRole("textbox", { name: "Value" }).first().fill("disorder");
+
+    // Set status
+    await page.getByRole("combobox", { name: "Status" }).click();
+    await page.getByRole("option", { name: "Active" }).click();
+
+    // Open preview and verify (as requested in review)
+    await page.getByRole("button", { name: "Value Preview" }).click();
```

Comment:
- In the preview search, use the faker array with 5 static value from the related system you used in the test to verify that the intended values are visible.